app/build.gradle.kts

```diff
@@ -121,6 +126,16 @@ android {
     }
 }
 
+tasks.register<ProvideVssDefinitionTask>("ProvideVssDefinition") {
+    val vssDefinitionFilePath = "$projectDir/src/main/assets/vss_rel_4.0.yaml"
+    val regularFile = RegularFile { File(vssDefinitionFilePath) }
+    vssDefinitionFile.add(regularFile)
+}
+
+tasks.withType<KspTask> {
+    dependsOn(tasks.withType<ProvideVssDefinitionTask>())
+}
+
```

Comment:
I would prefer to improve this so that the user does not need to add anything (except of the gradle plugin) if at all anything needs to be add, it should be "a well defined configuration" with properties similar to this here: https://github.com/JLLeitschuh/ktlint-gradle?tab=readme-ov-file#configuration

``` 
vss-processor { // absolutely no clue how we achieve the vss-processor scope but it would be awesome :D the configure would also be fine as long as it's optional
   vssSpecificationPath = "path/to/specification"
}
```


IMO the configuration could be optional and we simply crawl through the rootProject.fileTree() and search all yaml files, however this could create issues where different flavors use different specifications (if that is a valid use case)

Internally the plugin has a custom task: copySpecificationsToBuildFolder()
This task takes the configured vssSpecificationPath resp. all found .yaml files and copies them to where we need them

The plugin should automatically integrate the copySpecificationToBuildFolder task into the gradle lifecycle e.g. by
``` preBuild.dependsOn(copySpecificationsToBuildFolder) ```

Reply:
All configuration is now optional for the plugin. The default behavior is crawling through all compatible extension files (yaml) and copying to the KSP input folder. The "real" file name is then defined by the annotation anyway. The developer can still define a searchPath + fileName if they want via:

```
vssProcessor {
    searchPath = "$projectDir/src/main/assets"
    fileName = "vss_rel_4.0.yaml"
}
```

The task now depends on "preBuild".